testing-overhead/src/test/java/io/opentelemetry/results/ResultsCollector.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package io.opentelemetry.results;
+
+import com.jayway.jsonpath.JsonPath;
+import io.opentelemetry.util.JFRUtils;
+import io.opentelemetry.util.NamingConvention;
+import io.opentelemetry.results.AppPerfResults.MinMax;
+import io.opentelemetry.agents.Agent;
+import io.opentelemetry.config.TestConfig;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class ResultsCollector {
+
+  private final NamingConvention namingConvention;
+
+  public ResultsCollector(NamingConvention namingConvention) {this.namingConvention = namingConvention; }
+
+  public Map<Agent, AppPerfResults> collect(TestConfig config) {
```

Comment:
`AppPerfResults` already contains the `Agent`, why not return `List<AppPerfResults>` instead?

Or you could go a bit further and return sth like this:

```java
class AppPerfResults {
  TestConfig config;
  List<AgentResults> agentResults;
}

class AgentResults {
  Agent agent;
  // ... and all other fields from current AppPerfResults
}
```

Reply:
Changing the Map -> List worked out very nicely. I'm not ready to commit to the latter change yet tho 😁 , but I do see where you're coming from (the config is redundant).